samples/vsphere-disk-manager.py

```diff
@@ -0,0 +1,309 @@
+        print 'Destroying disk: ' + disk_path
+    else:
+        print 'Detaching disk: ' + disk_path
+    for d in disks:
+        if d.backing.fileName == disk_path:
+            detached_disk = d
+    if not detached_disk:
+        print disk_path + ' not attached'
+        sys.exit(1)
+    disk_spec = vim.vm.device.VirtualDeviceSpec()
+    disk_spec.operation = vim.vm.device.VirtualDeviceSpec.Operation.remove
+    if destroy:
+        disk_spec.fileOperation = \
+            vim.vm.device.VirtualDeviceSpec.FileOperation.destroy
+    disk_spec.device = detached_disk
+    spec = vim.vm.ConfigSpec()
+    spec.deviceChange = [disk_spec]
+    task = vm.ReconfigVM_Task(spec=spec)
+    wait_for_task(task, connection)
+    if destroy:
+        print 'Disk destroyed'
+    else:
+        print 'Disk detached'
+
+
+def destroy_disk(connection, vm, disks, data_store, disk_path):
+    detach_disk(connection, vm, disks, data_store, disk_path, True)
+
+
+def attach_disk(connection, vm, controller, available_unit_slot,
+                data_store, disk_path):
+    print "Attaching [%s] %s" % (data_store, disk_path)
+    spec = vim.vm.ConfigSpec()
+    dev_changes = []
+    disk_spec = vim.vm.device.VirtualDeviceSpec()
+    disk_spec.operation = vim.vm.device.VirtualDeviceSpec.Operation.add
+    disk_spec.device = vim.vm.device.VirtualDisk()
+    disk_spec.device.backing = vim.vm.device.VirtualDisk.FlatVer2BackingInfo()
+    disk_spec.device.backing.fileName = '[' + data_store + '] ' + disk_path
+    disk_spec.device.backing.diskMode = 'persistent'
+    disk_spec.device.backing.datastore = get_obj(connection, [vim.Datastore],
+                                                 data_store)
+    disk_spec.device.unitNumber = available_unit_slot
+    disk_spec.device.controllerKey = controller.key
+    dev_changes.append(disk_spec)
+    spec.deviceChange = dev_changes
+    task = vm.ReconfigVM_Task(spec=spec)
+    wait_for_task(task, connection)
+
+
+def wait_for_task(task, actionName='job', hideResult=False):
+    """
+    Waits and provides updates on a vSphere task
+    """
+    while task.info.state == vim.TaskInfo.State.running:
+        time.sleep(2)
+
+        if task.info.state == vim.TaskInfo.State.success:
+            if task.info.result is not None and not hideResult:
+                out = '%s completed successfully, result: %s' % \
+                    (actionName, task.info.result)
+                print out
+            else:
+                out = '%s completed successfully.' % actionName
+                print out
+        else:
+            out = '%s did not complete successfully: %s' % \
+                (actionName, task.info.error)
+            raise task.info.error
+            print out
+
+    return task.info.result
+
+
+def main():
+    args = get_args()
+    connection = connect(args)
+    vm = get_vm(args, connection)
+    print 'Devices:'
+    controller, available_unit_slot, disks = get_devices(vm)
+    if args.operation == 'add':
```

Comment:
Please add checks for required arguments if operation is add, destroy, attach & detach